roles/federation-demo/tasks/deploy.yml

```
@@ -0,0 +1,83 @@
+---
+  - name: Check if project exists
+    stat:
+      path: "{{ federation_git.clone_dest }}"
+    register: project_stat
+
+  - name: Clean clone directory
+    file:
+      state: absent
+      path: "{{ federation_git.clone_dest }}"
+    when: project_stat.stat.exists
+
+  - name: Clone the repository
+    git:
+      repo: "{{ federation_git.git_repo }}"
+      version: "{{federation_git.git_branch}}"
+      dest: "{{federation_git.clone_dest}}"
+      key_file: "{{ federation_git.key_path }}"
```

```suggestion
  - name: Check if project exists
    stat:
      path: "{{ federation_git.clone_dest }}"
    register: project_stat

  - name: Clean clone directory
    file:
      state: absent
      path: "{{ federation_git.clone_dest }}"
    when: project_stat.stat.exists

  - name: Clone the repository
    git:
      repo: "{{ federation_git.git_repo }}"
      version: "{{federation_git.git_branch}}"
      dest: "{{federation_git.clone_dest}}"
      key_file: "{{ federation_git.key_path }}"
      force: yes
```

If you want, you can check the [force](https://docs.ansible.com/ansible/2.9/modules/git_module.html#parameter-force)`:yes` parameter.